test/app-luatest/http_client_test.lua

```diff
@@ -654,3 +654,30 @@ g.test_concurrent = function(cg)
     t.assert(ok_sockets_added, "free sockets")
     t.assert(ok_active, "no active requests")
 end
+
+g.test_encode_body = function(cg)
```

Comment:
I'd also test a case when a table can't be json encoded.

E.g. try to pass a table that contains a function

Reply:
Yes. But happy paths are first ;)

Reply:
Wow, I didn't think about this, but when you said it, the voice in my head tells me: 'it is especially useful to detect a leak'. I looked at the revelant code and found the leak visually.